src/averageAges.js

```diff
@@ -15,13 +17,24 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const male = !century
+    ? personGender('m', people)
+    : personGender('m', people).filter(man =>
+      (century === Math.ceil(man.died / 100)));
+
+  return calculateAvarageAge(male);
 }
 
+const calculateAvarageAge = (people) => {
+  return people.reduce((sum, person) => (
+    sum + (person.died - person.born)), 0)
+  / people.length;
+};
+
+const personGender = (gender, people) => {
+  return people.filter(person => person.sex === gender);
+};
```

Comment:
you're lucky that tests work the way they does, remake it with function declaration, cause you're calling this function before they appear in code